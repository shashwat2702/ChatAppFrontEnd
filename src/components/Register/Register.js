import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Input from '../shared/Input/Input';
import Button from '../shared/Button/Button';
import './Register.scss';

export default class Register extends Component {
  render() {
    return (
        <div className="registrationForm">
      
          <ul class="tab-group">
            <li class="tab active"><Link to="/registration">Sign Up</Link></li>
            <li class="tab"><Link to="/login">Log In</Link></li>
          </ul>
      
        <div class="tab-content">
          <div id="signup">   
            <h1>Sign Up for Free</h1>
          
            <form action="/" method="post">
          
                <div class="field-wrap">
                <Input
                    id={ 'RegistrationUserName'}
                    label={'Username'}
                    className={'req'}
                    spanContent={'*'}
                    type={'text'}
                    placeholder={'Barack@president'}
                    autoComplete={'off'}
                />
                </div>
        
                <div class="field-wrap">
                  <Input
                    id={ 'RegistrationName'}
                    label={'Name'}
                    className={'req'}
                    spanContent={'*'}
                    type={'text'}
                    placeholder={'Barack Obama'}
                    autoComplete={'off'}
                />
                </div>
                <div class="field-wrap">
                  <Input
                    id={ 'RegistrationEmail'}
                    label={'Email ID'}
                    className={'req'}
                    spanContent={'*'}
                    type={'email'}
                    placeholder={'example@mail.com'}
                    autoComplete={'off'}
                />
                </div>
                <div class="field-wrap">
                 <Input
                  id={ 'RegistrationPassword'}
                  label={'Password'}
                  className={'req'}
                  spanContent={'*'}
                  type={'password'}
                  placeholder={'Password'}
                  autoComplete={'off'}
                />
                </div>    
              <Button type="submit" label={'Get Started'}/>
          
            </form>

          </div>
        
          <div id="login">   
          </div>
        </div>
      </div>
    )
  }
}
