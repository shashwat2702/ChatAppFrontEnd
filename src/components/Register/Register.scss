$body-bg: #c1bdba;
$form-bg: #13232f;
$white: #ffffff;

$main: #1ab188;
$main-light: lighten($main,5%);
$main-dark: darken($main,5%);

$gray-light: #a0b3b0;
$gray: #ddd;

$thin: 300;
$normal: 400;
$bold: 600;
$br: 4px;

*, *:before, *:after {
  box-sizing: border-box;
}

.registrationForm {
  background:rgba($form-bg,.9);
  padding: 40px;
  max-width:600px;
  margin:20px auto;
  border-radius:$br;
  box-shadow:0 4px 10px 4px rgba($form-bg,.3);
  a {
    text-decoration:none;
    color:$main;
  }
  .tab-group {
    list-style:none;
    padding:0;
    margin:0 0 40px 0;
    &:after {
      content: "";
      display: table;
      clear: both;
    }
    li a {
      display:block;
      padding:15px;
      background-color:rgba($gray-light,.25);
      color:$gray-light;
      font-size:20px;
      float:left;
      width:50%;
      text-align:center;
      cursor:pointer;
    }
    .active a {
      background:$main;
      color:$white;
    }
  }
  
  .tab-content > div:last-child {
    display:none;
  }
  
  
  h1 {
    text-align:center;
    color:$white;
    font-weight:$thin;
    margin:0 0 40px;
  }
  
    .req {
      margin:2px;
      color:$main;
    }
  
  .field-wrap {
    position:relative;
    margin-bottom:30px;
  }
  .errorBox {
    background-color: #ff9999;
    height: 30px;
    text-align: center;
    border: 1px solid #e60000;
    color: #862d2d;
    padding-top: 5px;
  }
  .successBox {
    background-color: #85e085;
    height: 30px;
    text-align: center;
    border: 1px solid #2aa22a;
    color: #155115;
    padding-top: 5px;
  }
}


