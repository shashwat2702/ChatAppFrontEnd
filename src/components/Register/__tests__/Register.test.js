import React from 'react';
import renderer from 'react-test-renderer';
import Register from '../Register';

jest.mock('react-router-dom');

describe('Login snapshot', () => {
  it('should render correctly', () => {
    const RegisterSnap = renderer.create(<Register />);
    const RegisterJson = RegisterSnap.toJSON();
    expect(RegisterJson).toMatchSnapshot();
  });
});
