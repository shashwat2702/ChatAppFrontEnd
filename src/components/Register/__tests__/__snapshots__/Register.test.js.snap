// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot should render correctly 1`] = `
<div
  className="registrationForm"
>
  <ul
    class="tab-group"
  >
    <li
      class="tab active"
    />
    <li
      class="tab"
    />
  </ul>
  <div
    class="tab-content"
  >
    <div
      id="signup"
    >
      <h1>
        Sign Up for Free
      </h1>
      <form
        action="/"
        method="post"
      >
        <div
          class="field-wrap"
        >
          <div
            className="inputComponent"
          >
            <label
              htmlFor="RegistrationUserName"
            >
              Username
              <span
                className="req"
              >
                *
              </span>
            </label>
            <input
              autoComplete="off"
              id="RegistrationUserName"
              placeholder="Barack@president"
              type="text"
            />
          </div>
        </div>
        <div
          class="field-wrap"
        >
          <div
            className="inputComponent"
          >
            <label
              htmlFor="RegistrationName"
            >
              Name
              <span
                className="req"
              >
                *
              </span>
            </label>
            <input
              autoComplete="off"
              id="RegistrationName"
              placeholder="Barack Obama"
              type="text"
            />
          </div>
        </div>
        <div
          class="field-wrap"
        >
          <div
            className="inputComponent"
          >
            <label
              htmlFor="RegistrationEmail"
            >
              Email ID
              <span
                className="req"
              >
                *
              </span>
            </label>
            <input
              autoComplete="off"
              id="RegistrationEmail"
              placeholder="example@mail.com"
              type="email"
            />
          </div>
        </div>
        <div
          class="field-wrap"
        >
          <div
            className="inputComponent"
          >
            <label
              htmlFor="RegistrationPassword"
            >
              Password
              <span
                className="req"
              >
                *
              </span>
            </label>
            <input
              autoComplete="off"
              id="RegistrationPassword"
              placeholder="Password"
              type="password"
            />
          </div>
        </div>
        <div
          className="buttonComponent"
        >
          <button
            className="button button-block"
            type="submit"
          >
            Get Started
          </button>
        </div>
      </form>
    </div>
    <div
      id="login"
    />
  </div>
</div>
`;
